src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java

```diff
@@ -581,4 +584,32 @@ public void testWithoutLogErrors() throws Exception {
                 CommonUtil.EMPTY_STRING_ARRAY);
     }
 
+    @Test
+    public void testClassAliasResolving() throws Exception {
+        final File file = new File(getPath("InputJavadocMethodClassAliasTest.java"));
+        final DetailAST rootAst = JavaParser.parseFile(file, JavaParser.Options.WITHOUT_COMMENTS);
+        final DetailAST classAst = rootAst.getNextSibling().getNextSibling();
+
+        final JavadocMethodCheck obj = new JavadocMethodCheck();
+        obj.beginTree(rootAst);
+        obj.visitToken(classAst);
+
+        final Class<?> clazz = Class.forName("com.puppycrawl.tools.checkstyle.checks.javadoc."
+                + "JavadocMethodCheck");
+        final Class<?> tokenClazz = Class.forName("com.puppycrawl.tools.checkstyle.checks.javadoc."
+                + "JavadocMethodCheck$Token");
+        final Constructor<?> tokenConstructor = tokenClazz.getDeclaredConstructor(String.class,
+                int.class, int.class);
+        final Object token = tokenConstructor.newInstance("B1", 21, 22);
+
+        final Method createClassInfo = clazz.getDeclaredMethod("createClassInfo", tokenClazz,
+                String.class);
+        createClassInfo.setAccessible(true);
+        final Object instanceResult = createClassInfo.invoke(obj, token,
+                "InputJavadocMethodClassAliasTest");
```

Comment:
Concern at https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-607237012

Reply:
@rnveach Replied in the issue comment thread.